#163355959 create the delete party endpoint

Chiazokam opened the pull request:
#### What does this PR do?
This PR tests and creates the delete party endpoint

#### Description of Task to be completed?
- write tests for all the functions
- create the endpoint

#### How should this be manually tested?
- Clone the repo - `https://github.com/Chiazokam/Politico`
- Enter into the project directory - `cd Politico`
- switch to this branch - `checkout ft-datastructures-delete-party-163355959`
- Start the server `npm run start:dev`
- Open `Postman`
- On `postman` Make a `DELETE` request to `http://localhost:3000/api/v1/parties/1 `
- You should receive a successful response

#### Any background context you want to provide?
Nil

#### What are the relevant pivotal tracker stories?
[#163355959](https://www.pivotaltracker.com/story/show/163355959)

####  Images

![image](https://user-images.githubusercontent.com/26940294/51785436-05d20780-2158-11e9-99fd-dadf1cb60ebc.png)

Chiazokam committed:
feat (frontend): set up the dev environment

- set up express
- set up babel
- set up travis
- set up coveralls

[Delivers #163377772]

Chiazokam committed:
ch (backend): setup node environment

- set up eslint
- set up travis
- set up coveralls
- set up mocha and chai

[Finishes #163418590]

Chiazokam committed:
feat (frontend): create the create party endpoint

- create the create party endpoint
- set up folder structure
- write tests for all functions
- design validators to check user input

[Delivers #163351807]

Chiazokam committed:
feat (frontend): create the create office endpoint

- write tests for all the functions
- create the endpoint
- create a validator to test the office name input
- create a validator to test the office type input

[Delivers #163356161]

Chiazokam committed:
feat (frontend): create the get all parties endpoint

- write tests for all the functions
- create the endpoint
- validate the existence of the party in create party

[Delivers #163351835]

Chiazokam committed:
feat (frontend): create the get all offices endpoint

- write tests for all the functions
- create the endpoint

[Delivers #163356189]

Chiazokam committed:
feat (frontend): create the get one party endpoint

- write tests for all the functions
- create the endpoint

[Delivers #163351844]

Chiazokam committed:
feat (frontend): create the get one office endpoint

- write tests for all the functions
- create the endpoint

[Delivers #163356516]

Chiazokam committed:
feat (frontend): create the edit a party endpoint

- write tests for all the functions
- create the endpoint

[Delivers #163351854]

Chiazokam committed:
feat (frontend): create the delete a party endpoint

- write tests for all the functions
- create the endpoint

[Delivers #163355959]

Chiazokam committed:
feat (frontend): create the delete a party endpoint

- write tests for all the functions
- create the endpoint

[Delivers #163355959]

Chiazokam committed:
Merge branch 'ft-datastructures-delete-party-163355959' of https://github.com/Chiazokam/Politico into ft-datastructures-delete-party-163355959

JCanaks commented on .babelrc:
```diff
@@ -0,0 +1,3 @@
+{
+    "presets": ["env"] 
+  }
```
Resolve all  EOL

JCanaks commented on server/controllers/officeControllers.js:
```diff
@@ -0,0 +1,69 @@
+                error: 'Office not created',
+            });
+    }
+    return res.status(400).send({
```
status `400` is not the appropriate status to use for already existing data

JCanaks commented on server/middlewares/officeValidators.js:
```diff
@@ -0,0 +1,52 @@
+    isOfficeFieldEmpty(req, res, next) {
+        const { officeName, officeType } = req.body;
+        const errors = {};
+        if (!officeName || !officeName.trim() || !officeType || !officeType.trim()) {
```
Why do we have the first if condition in line 19 that encloses others? is that necessary?

Chiazokam replied:
I put it there as the main error catching block. The other if statements are to populate the error array if any of them exist. Is that ok?

JCanaks commented on server/middlewares/partyValidators.js:
```diff
@@ -0,0 +1,62 @@
+  isPartyFieldEmpty(req, res, next) {
+    const { partyName, partyAddress, partyLogo } = req.body;
+    const errors = {};
+    if (!partyName || !partyName.trim() || !partyAddress || !partyAddress.trim() || !partyLogo || !partyLogo.trim()) {
```
Same here is the if condition in line 17 necessary?

JCanaks commented on server/middlewares/validateAddress.js:
```diff
@@ -0,0 +1,49 @@
+                    req.partyAddr = response.json.results[0].formatted_address;
+                    next();
+                } else {
+                    return res.status(404).send({
```
this line of code if unreachable when there is no result from the Google API `res.json.results` becomes an empty array --> `[]` checking if an empty array is defined (if (res.json.results) {...}) returns true so the else case is never captured and instead an error about formatted address pops up. its better to use  `if (res.json.results.length < 1) {}` this will capture the cases where there are no results

JCanaks commented on server/helpers/partyHelpers.js:
```diff
@@ -0,0 +1,63 @@
+
+  createParty(party) {
+    const newParty = {
+      id: this.parties.length + 1,
```
Pls use the field Name format as specified in the ADC docs 
<img width="544" alt="screen shot 2019-01-29 at 12 59 59 am" src="https://user-images.githubusercontent.com/26818808/51877037-ebbc4300-236a-11e9-8b61-6499fcf77d38.png">

JCanaks commented on server/middlewares/partyValidators.js:
```diff
@@ -0,0 +1,62 @@
+    next();
+  }
+
+  isPartyNameString(req, res, next) {
```
@Chiazokam Where is the check for when an interger is entered as a `partyName` or `partyAddress`?

Chiazokam committed:
feat (backend): implement feedback from LFA

- validate for where address is a number
- change variable names to specified names
- capture cases from google api where array is empty
- write tests for all introduced functions
- include EOL

[Delivers #163355959]
